storefront/test/integration/workarea/storefront/recent_views_integration_test.rb

```diff
@@ -0,0 +1,40 @@
+require 'test_helper'
+
+module Workarea
+  module Storefront
+    class RecentViewsIntegrationTest < Workarea::IntegrationTest
+      include Storefront::IntegrationTest
+
+      def test_show
+        user = create_user
+        category = create_category
+        href = storefront.product_path(product, via: category.id)
+
+        sign_in(user)
+        Metrics::User.save_affinity(
+          id: user.email,
+          action: 'viewed',
+          product_ids: [product.id]
+        )
+        get storefront.recent_views_path(via: category.id)
+
+        assert_response(:success)
+        assert_select('.recent-views__section--products')
+        assert_select('.product-summary')
+        assert_select(%(.product-summary__name a[href="#{href}"]))
+      end
+
+      private
+
+      def sign_in(user, password = 'W3bl1nc!')
```

Comment:
You'll want to specify this password to `create_user` in case something changes. Also not sure a separate method is really adding much here given this is only used once.